x-pack/plugins/security_solution/public/common/components/ml/links/create_explorer_link.tsx

```diff
@@ -22,26 +22,46 @@ export const ExplorerLink: React.FC<ExplorerLinkProps> = ({
   endDate,
   linkName,
 }) => {
-  const { getUrlForApp } = useKibana().services.application;
+  const urlGenerator = useKibana().services.ml?.urlGenerator;
+  const [explorerUrl, setExplorerUrl] = useState('');
+
+  useEffect(() => {
```

Comment:
minor nit: we have a `useMlLink` hook that you can use here

Reply:
@qn895  It's not possible to reuse this hook because the Security App has a different Kibana context, also `share` is not available. We also have `useMlHref`, but I don't follow why it requires a `basePath`.

Reply:
Updated in https://github.com/elastic/kibana/pull/83792/commits/db3d13789a4fb5e19df04ab427e9fb7d17ff5e7a

Reply:
Thanks @darnautov. I think `useMlHref` requires a basePath because of limitations from other plugins at the time (I think they redirect with full page reload without the basePath iirc). This change LGTM.